clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -250,6 +250,20 @@ static bool CheckSYCLKernelName(Sema &S, SourceLocation Loc,
   return false;
 }
 
+void SemaSYCL::CheckSYCLExternalFunctionDecl(FunctionDecl *FD) {
+  const auto *SEAttr = FD->getAttr<SYCLExternalAttr>();
+  assert(SEAttr && "Missing sycl_external attribute");
+  if (!FD->isExternallyVisible()) {
+    Diag(SEAttr->getLocation(), diag::err_sycl_attribute_invalid_linkage);
+    return;
+  }
```

Comment:
An early return should not be done here since that would prevent other possible diagnostics.
```suggestion
  if (!FD->isExternallyVisible())
    Diag(SEAttr->getLocation(), diag::err_sycl_attribute_invalid_linkage);
```